### DISCUSSION TITLE
Django 3.2 + Python 3.7

### DISCUSSION THREAD

**Discussion by ewjoachim** [codingjoe/django-select2#197] [Q&A] [answered]
**Goal**
We're on our way upgrading a codebase from an old Django to the latest LTS. We're currently on Python 3.7 and it will be complicated to change both python and django at the same time

**Problem**
As far as the metadata tell
- In [7.10.0](https://github.com/codingjoe/django-select2/blob/7.10.0/setup.cfg#L17-L26), the lib was compatible with py3.7 but not with django 3.2
- In [7.10.1](https://github.com/codingjoe/django-select2/blob/7.10.1/setup.cfg#L17-L25), the lib was not (officially) compatible with py3.7, but was compatible with django 3.2

That said, I'm inclined to think it's an error in the metadata, because, for example, [this commit](https://github.com/codingjoe/django-select2/commit/e15ced997d376f986f4cb6c77eb70846733d7265) added compatibility to django 4.0 while python 3.7 was still supported, and it's strange that the lib would be compatible with django 3.1 and 4.0 and **not** 3.2.

I think I can try to find what gives exactly, but if you already know the issue and you can tell from the top of your head what version we can target to have both py3.7 and django3.2 without it beeing a hassle, you can save me a headache :)

(I'd be inclined to take 7.10.0 as the last version that officially supported py3.7 and assume it supported dj3.2, even though it looks like nothing was changed in subsequent commits that would likely make later versions fail with py3.7)

Thanks in advance !

**Comment by ewjoachim** [comment]
Yeah, I see 3.2 was added to the test suite in https://github.com/codingjoe/django-select2/commit/038026c2f96d41810de567a7c159655c30bee46f so 7.7.1, so that 99% answers the question. If you can just confirm that it was a small metadata issue and not a bigger problem, I won't bother you anymore :)

**Comment by codingjoe** [comment, marked as answer]
Hi @ewjoachim,

Thank you for reaching out. I understand how upgrading from a legacy version isn't all too fun.
This package is very mature and rather stable in terms of features and code base. Thought, we do upgrade your CI suite and version markers, code largely remains the same. Therefore, you'll probably find mother version compatible with older dependencies.

However, I would always encourage anyone to upgrade Django versions frequently. Django has been getting pretty stable and there is tooling now, that automates upgrades, like `pyupgrade` and `django-upgrade`.

I hope this help you a little.

Cheers!
Joe

**Comment by ewjoachim** [comment]
> I would always encourage anyone to upgrade Django versions frequently

Yep, that's exactly the plan, but the first step is to get to a supported LTS, thus my question :)

Thanks for your answer.
